### PR TITLE
Self documentation of CLI

### LINKED ISSUE #124: Unify the CLI output
Today we have a bit-messy output on the command-line. we should unify the way we print information on the output.

* `init` / `service init` use a logger
* other commands use basic output printing.
* there is no structure, we need to establish a clear structure for deprecation message, warnings and kind to stand out.

### LINKED ISSUE #125: Support multiple output format on the CLI
Supporting multiple output format such as text and JSON is increasingly popular on CLI that might be worth investigating.

### LINKED ISSUE #126: Add some example / explanation on the command line to learn how to do stuff.
We should ty to make the CLI of wallet kind of self documented.

For example, if the use just ran `vega init`, we could have this kind of output:
```Shell
$ vega init
The vega wallet home has been created at /home/user/bob

    Now that you have initialised the wallet you may want to either create a new keypair:
    > vega wallet key generate --name="USER_NAME"

    or event start the wallet server to use it with the console:
    > vega wallet service run -p
```

### LINKED ISSUE #199: Require user confirmation if running unreleased version
consider baking a protective mechanism into CLI if someone complies directly from develop - “are you sure you want to use this non-stable version of the wallet? If you want to carry on, enter the command ‘I accept this risk’, or get the tested/stable version from (link)”

### PR DESCRIPTION
**Pull request by ValentinTrinque**
To make the CLI more user-friendly, I updated the output to a rich, informative anb colorful output nicknamed "human" output. However this output is not script-friendly, as a result, I introduced an `--output` flag with a JSON output.

List of changes:
1. Remove the `--passphrase` flag to only have the `--passphrase-file` one. If no TTY is detected, a missing `--passphrase-file` results in a failing script instead of a hanging blank output.
2. Remove the non long-deprecated `service init` sub-command
3. Introduction of the JSON output, that prints data as JSON on a single line. Some commands print nothing. The success and failure is determined by the code returned by the command.
4. Introduction of the _Human_  output, that prints some logs, important notes and next steps. I added colors to enhance text structure.

@candida-d 
* Could you please try out the new "human" output ? 
* Could you also check the documentation is aligned with the list of changes above ?

@ashleyvega 
* Could you give me some feedback about the `--output=json` flag and how it maps to your scripts ?

Close #126 
Close #124
Close #125
Close #199

### CODE REVIEW

**Comment by ashleyvega** [comment]
@ValentinTrinque I have checked DV and it only uses `go-wallet init`, all other interaction is done with the REST API. The hosted wallets are the same - I ran `go-wallet init` ages ago, then everything else is done via REST.

Minor non-blocking note: I notice that `go-wallet service run` still produces text (e.g. Available endpoints) then json (e.g. `{"level":"error","message":"error starting wallet http server","error":"listen tcp 127.0.0.1:1789: bind: address already in use"}`, rather than text xor json.

**Comment by ValentinTrinque** [comment]
@ashleyvega  I was resistant on this for some reasons but think I made something ok. If you mind to test it.

File: cmd/root.go [hunk 1]
```diff
@@ -7,64 +7,113 @@ import (
...
+	}
+
+	return fmt.Errorf("unsupported output \"%s\"", rootArgs.output)
+}
+
+func checkVersion() {
 	if !rootArgs.noVersionCheck {
 		v, err := version.Check(version.Version)
 		if err != nil {
 			fmt.Printf("could not check vega wallet version updates: %v\n", err)
```
Comment (candida-d) [inline cmd/root.go:70]:
```suggestion
			fmt.Printf("could not check Vega wallet version updates: %v\n", err)
```

File: cmd/root.go [hunk 2]
```diff
@@ -7,64 +7,113 @@ import (
...
 		}
 		if v != nil {
-			fmt.Printf("A new version %v of vega wallet is available, you can download it at %v.\n",
-				v, version.GetReleaseURL(v))
+			p := termenv.ColorProfile()
+			xVersion := termenv.String(v.String()).Foreground(p.Color("6"))
+			xURL := termenv.String(version.GetReleaseURL(v)).Underline()
+			fmt.Printf("A new version %s of vega wallet is available!\nDownload it at %v\n\n", xVersion, xURL)
```
Comment (candida-d) [inline cmd/root.go:76]:
```suggestion
			fmt.Printf("A new version %s of Vega wallet is available!\nDownload it at %v\n\n", xVersion, xURL)
```

**Comment by candida-d** [comment]
Generally speaking, the instructions are really clear and provide a lot more guidance, definitely less need to rely on external instructions once you've actually started it up so it hits that goal. 

* The mnemonic here is the only information that you need and can't retrieve, so I think it should be visually more important than the rest of the details - I think it's too easy to gloss over this and miss the opportunity to save it. 
* Is it necessary (and secure?) to expose the private key here? 
Screenshot 2021-08-30 at 17 47 03

**Comment by candida-d** [comment]
I had a thought that might be bigger than this PR. If people will be able to use this wallet for their real Vega tokens, is there anything odd about linking to Vega Console? Can the wallet interact with testnet assets and real assets without causing any confusion for users?

**Review by candida-d — COMMENTED** [review]
I capitalised Vega where I saw it lowercase but there might be a few in other instructions that I missed. 

I also left some feedback about the mnemonic in a comment.

**Comment by ValentinTrinque** [comment]
* Remove some info from `key generate` command. Keep public key, metadata and mnemonic.
* "You should write your mnemonic on a paper"
* "Do not share you private key nor you mnemonic."

**Review by jeremyletang — DISMISSED** [review]

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**